File: ios/StarterApp/Core/Sync/ConflictResolver.swift
import Foundation

enum ResolutionStrategy {
    case serverWins
    case clientWins
    case mostRecent
    case merge
    case manual
}

struct AnyConflict {
    let id: String
    let collection: String
    let conflict: SyncConflict
}

struct ConflictResolver {
    static func resolve(_ conflict: AnyConflict, strategy: ResolutionStrategy) -> ResolutionStrategy {
        switch strategy {
        case .mostRecent:
            // Compare timestamps; caller applies chosen version
            return .mostRecent
        default:
            return strategy
        }
    }
}

File: android/app/src/main/kotlin/com/onlystack/starterapp/core/sync/ConflictResolver.kt
package com.onlystack.starterapp.core.sync

sealed class ResolutionStrategy {
    object ServerWins : ResolutionStrategy()
    object ClientWins : ResolutionStrategy()
    object MostRecent : ResolutionStrategy()
    object Merge : ResolutionStrategy()
    object Manual : ResolutionStrategy()
}

object ConflictResolver {
    fun resolve(conflict: SyncConflict, strategy: ResolutionStrategy): ResolutionStrategy {
        return when (strategy) {
            is ResolutionStrategy.MostRecent -> ResolutionStrategy.MostRecent
            else -> strategy
        }
    }
}
